File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

import javax.lang.model.element.Element;

class Solution {
    boolean solution(String s) {
        Stack<Integer> stack = new Stack<>(); //push, pop, peek, empty, seach 지원
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                stack.push(1);
            } else {
                if (stack.empty()) {
                    return false;
                } else {
                    stack.pop();
                }
            }
        }
        if (stack.empty()) {
            return true;
        } else {
            return false;
        }
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    st = []
    s = list(s)
    for i in s:
        if i == "(":
            st.append(1)
        elif st == []:
            return False
        else:
            st.pop()
    
    if st == []:
        return True
    else:
        return False